src/test/java/io/confluent/examples/streams/IntegrationTestUtils.java

```diff
@@ -277,4 +279,87 @@ private static boolean continueConsuming(final int messagesConsumed, final int m
         30000,
         "Expected values not found in WindowStore"); }
 
+  static <K, V> Map<K, V> drainTableOutput(final String outputTopic,
+                                           final TopologyTestDriver topologyTestDriver,
+                                           final Deserializer<K> keyDeserializer,
+                                           final Deserializer<V> valueDeserializer) {
+
+    final Map<K, V> results = new LinkedHashMap<>();
+    while (true) {
+      final ProducerRecord<K, V> record = topologyTestDriver.readOutput(outputTopic, keyDeserializer, valueDeserializer);
+      if (record == null) {
+        break;
+      } else {
+        results.put(record.key(), record.value());
+      }
+    }
+    return results;
+  }
+
+  static <K, V> void produceKeyValuesSynchronously(final String inputTopic,
+                                                   final List<KeyValue<K, V>> entities,
+                                                   final TopologyTestDriver topologyTestDriver,
+                                                   final Serializer<K> keySerializer,
+                                                   final Serializer<V> valueSerializer) {
+    for (final KeyValue<K, V> entity : entities) {
+      topologyTestDriver.pipeInput(new ConsumerRecord<>(
+        inputTopic,
+        0,
+        0,
+        0,
+        TimestampType.CREATE_TIME,
+        -1,
+        -1,
+        -1,
+        keySerializer.serialize(inputTopic, entity.key),
+        valueSerializer.serialize(inputTopic, entity.value)
+      ));
+    }
+  }
+
+  /**
+   * Creates a map entry (for use with {@link IntegrationTestUtils#mkMap(java.util.Map.Entry[])})
+   *
+   * @param k   The key
+   * @param v   The value
+   * @param <K> The key type
+   * @param <V> The value type
+   * @return An entry
+   */
+  static <K, V> Map.Entry<K, V> mkEntry(final K k, final V v) {
+    return new Map.Entry<K, V>() {
+      @Override
+      public K getKey() {
+        return k;
+      }
+
+      @Override
+      public V getValue() {
+        return v;
+      }
+
+      @Override
+      public V setValue(final V value) {
+        throw new UnsupportedOperationException();
+      }
+    };
+  }
+
+  /**
+   * Creates a map from a sequence of entries
+   *
+   * @param entries The entries to map
+   * @param <K>     The key type
+   * @param <V>     The value type
+   * @return A map
+   */
+  @SafeVarargs
+  static <K, V> Map<K, V> mkMap(final Map.Entry<K, V>... entries) {
+    final LinkedHashMap<K, V> result = new LinkedHashMap<>();
```

Comment:
Why `LinkedHashMap`?

Reply:
Same basic reason: The API is Map, so I'm not guaranteeing anything, but I figured it would help test/debugging ergonomics if the order were stable and depends on the code to create the map.

Reply:
Ack. Should it be `Map` left hand side?

Reply:
Sure!